src/main/java/com/fasterxml/jackson/databind/util/LookupCache.java

```diff
@@ -0,0 +1,29 @@
+package com.fasterxml.jackson.databind.util;
+
+import com.fasterxml.jackson.core.util.Snapshottable;
+
+import java.util.function.BiConsumer;
+
+/**
+ * An interface describing the required API for the Jackson-Databind Type cache.
+ *
+ * @see com.fasterxml.jackson.databind.type.TypeFactory#withCache(LookupCache<java.lang.Object,com.fasterxml.jackson.databind.JavaType>)
+ * @see SimpleLookupCache
+ */
+public interface LookupCache <K,V>
+        extends Snapshottable<LookupCache<K,V>>,
+        java.io.Serializable {
```

Comment:
Not sure `java.io.Serializable` is actually needed for interface itself.
Serializability in 3.0 is by serializing Builders. Although it may be true that for some types, implementations would need to be serializable. Snapshottable does make sense, I think.